src/shared/api/api.js

```diff
@@ -90,24 +90,30 @@ function graphql({
       variables,
     }),
   })
-    .then(async (res) =>
-      res.ok
-        ? await res.json()
-        : Promise.reject({
-            status: res.status,
-            data: await res.json(),
-          })
-    )
-    .catch((err) => {
-      if (err.status === 401 && config.IS_SELF_HOSTED) {
-        window.location.href = '/login'
+    .then(async (res) => {
+      const data = await res.json()
+
+      if (data?.errors) {
+        if (
+          data?.errors?.[0]?.extensions?.status === 403 &&
+          config.IS_SELF_HOSTED
+        ) {
+          window.location.href = '/login'
+        }
+      }
```

Comment:
I just want to double confirm here, users who are logged in, but maybe not have access to something let's say the admin panel in self hosted (which users REST endpoints) won't get redirect to the login page right?

Reply:
Sorry I totally missed that this change is only occurring in the `graphqil` method 🤦